### PR TITLE
Add type annotations to all public methods in `support.bits`

### PR DESCRIPTION
**Pull request by whitequark**
Is this a policy change? Yes: I'm looking to gradually add typing annotations (specifically the Pylance dialect) to Glasgow, and then eventually to Amaranth.

### CODE REVIEW

**Comment by whitequark** [comment]
... Oh, we can't do this until the lowest Python version we support is 3.11.

**Comment by whitequark** [comment]
Or start depending on https://pypi.org/project/typing-extensions/.

File: software/glasgow/support/bits.py [hunk 1]
```diff
@@ -136,18 +139,18 @@ def __new__(cls, value=0, length=None):
             return cls.from_iter(value)
         raise TypeError(f"invalid input for {cls.__name__}(): cannot convert from {value.__class__.__name__}")
 
-    def __len__(self):
+    def __len__(self )-> int:
```
Comment (wanda-phi) [inline software/glasgow/support/bits.py:142]: whitespace error

**Comment by wanda-phi** [comment]
The `bits` annotations look OK; I can't say I like Python type annotations much (they are a gross hack tbh), but I guess it's better than nothing.

**Pull request merged**